core/src/main/java/org/jboss/pnc/rex/core/GenericVertxHttpClient.java

```diff
@@ -148,17 +146,15 @@ private Uni<HttpResponse<Buffer>> handleRequest(Uni<HttpResponse<Buffer>> uni, C
                 );
 
         // cases when http request method itself fails (unreachable host)
-        uni = uni.onFailure(this::skipOn425)
+        uni = uni.onFailure(this::skipOnResponse)
                 .invoke(t ->  {
                     if (t instanceof RequestRetryException) {
                         log.warn("HTTP-CLIENT : Http call failed. RETRYING.");
-                    } else {
-                        log.warn("HTTP-CLIENT : Http call failed. RETRYING. Reason: ", t);
+                    } else if (t instanceof HttpResponseException) {
+                        log.warn("HTTP-CLIENT : Http call failed ({}). RETRYING.", ((HttpResponseException) t).getStatusCode());
```

Comment:
Should be no need to cast because of this https://docs.oracle.com/en/java/javase/17/language/pattern-matching-instanceof.html